src/poll.rs

```diff
@@ -1,74 +1,48 @@
-use libc::c_int;
+use libc;
 use {Errno, Result};
 
-pub use self::ffi::PollFd;
-pub use self::ffi::consts::*;
-
-mod ffi {
-    use libc::c_int;
-    pub use self::consts::*;
-
-    #[derive(Clone, Copy, Debug)]
-    #[repr(C)]
-    pub struct PollFd {
-        pub fd: c_int,
-        pub events: EventFlags,
-        pub revents: EventFlags
-    }
-
-    #[cfg(target_os = "linux")]
-    pub mod consts {
-        use libc::{c_short, c_ulong};
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct PollFd {
+    pollfd: libc::pollfd,
+}
 
-        bitflags! {
-            flags EventFlags: c_short {
-                const POLLIN     = 0x001,
-                const POLLPRI    = 0x002,
-                const POLLOUT    = 0x004,
-                const POLLRDNORM = 0x040,
-                const POLLWRNORM = 0x100,
-                const POLLRDBAND = 0x080,
-                const POLLWRBAND = 0x200,
-                const POLLERR    = 0x008,
-                const POLLHUP    = 0x010,
-                const POLLNVAL   = 0x020,
-            }
+impl PollFd {
+    pub fn new(fd: libc::c_int, events: EventFlags, revents: EventFlags) -> PollFd {
+        PollFd {
+            pollfd: libc::pollfd {
+                fd: fd,
+                events: events.bits(),
+                revents: revents.bits(),
+            },
         }
-
-        pub type nfds_t = c_ulong;
     }
 
-    #[cfg(target_os = "macos")]
-    pub mod consts {
-        use libc::{c_short, c_uint};
-
-        bitflags! {
-            flags EventFlags: c_short {
-                const POLLIN     = 0x0001,
-                const POLLPRI    = 0x0002,
-                const POLLOUT    = 0x0004,
-                const POLLRDNORM = 0x0040,
-                const POLLWRNORM = 0x0004,
-                const POLLRDBAND = 0x0080,
-                const POLLWRBAND = 0x0100,
-                const POLLERR    = 0x0008,
-                const POLLHUP    = 0x0010,
-                const POLLNVAL   = 0x0020,
-            }
-        }
-
-        pub type nfds_t = c_uint;
+    pub fn revents(&self) -> Option<EventFlags> {
```

Comment:
I added these bindings to nix in #228 as I planned to use it in a project but I never got around to it since then. A [quick code search](https://github.com/search?utf8=%E2%9C%93&q=%22nix%3A%3Apoll%22+language%3Arust&type=Code&ref=searchresults) on Github shows atleast 2 people are using it: https://github.com/tailhook/stator and https://github.com/hjr3/carp-rs; they might have some useful feedback :)